helpers/target.go

```diff
@@ -377,107 +378,170 @@ func IsAWSAccReachable(accARN, assumeRoleURL, role string, sessDuration int) (bo
 		assumeRoleResp.SessionToken), nil
 }
 
-// IsDockerImgReachable returns wether the input Docker image exists
-// in registry. Void user and pass does not produce an error and will
-// interact with registry without authentication.
+// IsDockerImgReachable returns whether the input Docker image exists in the
+// registry. Void user and pass does not produce an error unless a token can be
+// generated without authentication.
+// In order to verify if the Docker image exists, we perform a request to
+// registry API endpoint to get data for given image and tag.
+//
+// This functionality at the moment of this writing is still not
+// implemented in Docker client, so we have to contact registry's REST API
+// directly.  Reference: https://github.com/moby/moby/issues/14254
 func IsDockerImgReachable(target, user, pass string) (bool, error) {
 	repo, err := parseDockerRepo(target)
 	if err != nil {
 		return false, err
 	}
 
-	// In order to verify if Docker img exists, we perform
-	// a request to registry API endpoint to get data for
-	// given image and tag.
-	//
-	// This functionality at the moment of this writing is
-	// still not implemented in Docker client, so we have
-	// to contact registry's REST API directly.
-	// Reference: https://github.com/moby/moby/issues/14254
+	token, err := dockerAPIToken(repo, user, pass)
+	if err != nil {
+		return false, err
+	}
 
-	// Check registry version.
-	// Reference: https://docs.docker.com/registry/spec/api/#api-version-check
-	regVersion := "v1"
-	resp, err := http.Get(fmt.Sprintf("https://%s/v2/", repo.Registry))
+	// Check there exist tags for the image.
+	regBase := fmt.Sprintf("https://%s/v2/", repo.Registry)
+	tagPath := fmt.Sprintf("%s%s/%s/%s/", regBase, repo.Img, "tags", "list")
```

Comment:
Why not use format as _"%s%s/tags/list"_ ?

Reply:
Yeah, that was tricky. In that case the URL generated was something like https://registry/v2//image/tags/list and that extra double slash was making the token generated before not work.

Reply:
Re-reading your comment I was answering to a different and unformulated question xDDD

You are right, I'm gonna fix it.